uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -559,71 +414,87 @@ define([
-          }.bind(this));
-      } else {
-        return $q.resolve();
-      }
-    }
-
-    /**
      * Initialises listeners
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::requestObjectUpdated', setInitialAttributes.bind(this)),
-        $rootScope.$on('LeaveRequestPopup::childComponent::register', function () { childComponentsCount++; })
+        $rootScope.$on('LeaveRequestPopup::requestObjectUpdated', setInitialAttributes),
+        $rootScope.$on('LeaveRequestPopup::handleError', function (__, errors) {
+          handleError(errors);
+        }),
+        $rootScope.$on('LeaveRequestPopup::childComponent::register', function () {
+          childComponentsCount++;
+        })
       );
     }
 
     /**
      * Initialises open mode of the dialog
      */
     function initOpenMode () {
-      if (this.request.id) {
-        this.mode = 'edit';
+      if (vm.request.id) {
+        vm.mode = 'edit';
 
         var viewModeStatuses = [
-          this.requestStatuses[sharedSettings.statusNames.approved].value,
-          this.requestStatuses[sharedSettings.statusNames.adminApproved].value,
-          this.requestStatuses[sharedSettings.statusNames.rejected].value,
-          this.requestStatuses[sharedSettings.statusNames.cancelled].value
+          vm.requestStatuses[sharedSettings.statusNames.approved].value,
+          vm.requestStatuses[sharedSettings.statusNames.adminApproved].value,
+          vm.requestStatuses[sharedSettings.statusNames.rejected].value,
+          vm.requestStatuses[sharedSettings.statusNames.cancelled].value
         ];
 
-        if (this.isRole('staff') && viewModeStatuses.indexOf(this.request.status_id) > -1) {
-          this.mode = 'view';
+        if (vm.isRole('staff') && viewModeStatuses.indexOf(vm.request.status_id) > -1) {
+          vm.mode = 'view';
         }
       } else {
-        this.mode = 'create';
+        vm.mode = 'create';
+      }
+    }
+
+    /**
+     * Initializes the leave request object
+     */
+    function initRequest () {
+      vm.request = vm.leaveRequest || null;
+      var leaveType = getLeaveType();
+      var attributes = vm.initRequestAttributes();
+
+      if (leaveType === 'leave') {
+        vm.request = LeaveRequestInstance.init(attributes);
+      } else if (leaveType === 'sickness') {
+        vm.request = SicknessRequestInstance.init(attributes);
+      } else if (leaveType === 'toil') {
+        vm.request = TOILRequestInstance.init(attributes);
+      }
+    }
+
+    /**
+     * Initialize request attributes based on directive
+     *
+     * @return {Object} attributes
+     */
+    function initRequestAttributes () {
+      var attributes = {};
+
+      // if set indicates self leave request is either being managed or edited
+      if (vm.request) {
+        // _.deepClone or angular.copy were not uploading files correctly
```

Comment:
This comment is confusing, there is no _.deepClone anywhere. Maybe is it old and we should delete it?